packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -47,8 +47,11 @@ export interface TerminalWidgetFactoryOptions extends Partial<TerminalWidgetOpti
 export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget {
 
     private readonly TERMINAL = 'Terminal';
+    isInteractive = true;
+    terminalId = -1;
     protected readonly onTermDidClose = new Emitter<TerminalWidget>();
-    protected terminalId = -1;
```

Comment:
Why to expose to clients?

Reply:
because we exposed the terminalId in TaskInfo interface, i.e.,  `TaskInfo.terminalId: number`. And we need it from the terminalWidget to find the association between the running tasks and widgets after the browser is refreshed.

https://github.com/eclipse-theia/theia/pull/7260/files#diff-38e65dfb04ff1bf292adf8abf4e1b9d5R92-R96